packages/statemanager/src/statefulVerkleStateManager.ts

```diff
@@ -0,0 +1,467 @@
+import type { VerkleCrypto } from '@ethereumjs/util'
+import type { Debugger } from 'debug'
+export class StatefulVerkleStateManager implements StateManagerInterface {
+  protected _debug: Debugger
+  protected _caches?: Caches
+
+  originalStorageCache: OriginalStorageCache
+
+  protected _trie: VerkleTree
+
+  public readonly common: Common
+
+  protected _checkpointCount: number
+
+  protected verkleCrypto: VerkleCrypto
+  /**
+   * StateManager is run in DEBUG mode (default: false)
+   * Taken from DEBUG environment variable
+   *
+   * Safeguards on debug() calls are added for
+   * performance reasons to avoid string literal evaluation
+   * @hidden
+   */
+  protected readonly DEBUG: boolean = false
+
+  private keccakFunction: Function
+
+  constructor(opts: StatefulVerkleStateManagerOpts) {
+    // Skip DEBUG calls unless 'ethjs' included in environmental DEBUG variables
+    // Additional window check is to prevent vite browser bundling (and potentially other) to break
+    this.DEBUG =
+      typeof window === 'undefined' ? (process?.env?.DEBUG?.includes('ethjs') ?? false) : false
+
+    this._checkpointCount = 0
+
+    if (opts.common?.isActivatedEIP(6800) === false)
+      throw new Error('EIP-6800 required for verkle state management')
+
+    this.common = opts.common ?? new Common({ chain: Mainnet, eips: [6800] })
+    this._trie =
+      opts.trie ??
+      new VerkleTree({ verkleCrypto: opts.verkleCrypto, db: new MapDB<Uint8Array, Uint8Array>() })
+    this._debug = debugDefault('statemanager:statefulVerkleStatemanager')
+    this.originalStorageCache = new OriginalStorageCache(this.getStorage.bind(this))
+    this._caches = opts.caches
+    this.keccakFunction = opts.common?.customCrypto.keccak256 ?? keccak256
+    this.verkleCrypto = opts.verkleCrypto
+  }
+
+  getAccount = async (address: Address): Promise<Account | undefined> => {
+    const elem = this._caches?.account?.get(address)
+    if (elem !== undefined) {
+      return elem.accountRLP !== undefined
+        ? createPartialAccountFromRLP(elem.accountRLP)
+        : undefined
+    }
+
+    const stem = getVerkleStem(this.verkleCrypto, address, 0)
+
+    // First retrieve the account "header" values from the trie
+    const accountValues = await this._trie.get(stem, [
+      VerkleLeafType.BasicData,
+      VerkleLeafType.CodeHash,
+    ])
+
+    let account
+    if (accountValues[0] !== undefined) {
+      const basicData = decodeVerkleLeafBasicData(accountValues[0]!)
+      account = createPartialAccount({
+        version: basicData.version,
+        balance: basicData.balance,
+        nonce: basicData.nonce,
+        codeHash:
+          accountValues[1] === undefined
+            ? KECCAK256_NULL // codeHash is undefined in trie (i.e. not touched)
+            : equalsBytes(accountValues[1], new Uint8Array(32))
+              ? KECCAK256_NULL // codeHash is deleted in trie (i.e. overwritten with zeroes)
+              : accountValues[1],
```

Comment:
I think this is equivalent and imo easier to parse: 
```suggestion
        codeHash:
          accountValues[1] === undefined || equalsBytes(accountValues[1], new Uint8Array(32)) ? KECCAK256_NULL : accountValues[1]
```

Reply:
Good call.  Thanks!